File: backend/JMMinistry.Common/Dtos/User/TokenResult.cs
namespace JMMinistry.Common.Dtos.User
{
    public class TokenResult
    {
        public bool IsAuthenticated { get; set; } = false;
        public DateTime Expiration { get; set; }
        public string Token { get; set; } = string.Empty;
        public string RefreshToken { get; set; } = string.Empty;
    }
}

File: the_vine_hub_app/lib/features/auth/domain/token_result.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'token_result.freezed.dart';
part 'token_result.g.dart';

@freezed
abstract class TokenResult with _$TokenResult {
  const factory TokenResult({
    required bool isAuthenticated,
    required DateTime expiration,
    required String token,
    required String refreshToken,
  }) = _TokenResult;

  factory TokenResult.fromJson(Map<String, dynamic> json) =>
      _$TokenResultFromJson(json);
}
